src/lib/wallet/IndyWallet.ts

```diff
@@ -69,6 +69,13 @@ export class IndyWallet implements Wallet {
     return this.indy.issuerCreateAndStoreCredentialDef(this.wh, issuerDid, schema, tag, signatureType, config);
   }
 
+  public async createCredentialOffer(credDefId: CredDefId) {
+    if (!this.wh) {
+      throw Error('Wallet has not been initialized yet');
+    }
```

Comment:
I'm not a fan of this function also added to the wallet. Basically the same problem as with create credential definition....

Reply:
Yeah, unfortunately. I got an idea to have more objects according to their responsibility, each with reference to the same `wh` which could be passed during the creation of such objects by a builder.